lib/client-assets.php

```diff
@@ -834,6 +834,18 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
 	// to disable it outright.
 	wp_enqueue_script( 'heartbeat' );
 
+	/**
+	 * Fires after block assets have been enqueued for the editing interface.
+	 *
+	 * Call `add_action` on any hook before 'admin_enqueue_scripts'.
+	 *
+	 * In the function call you supply, simply use `wp_enqueue_script` and
+	 * `wp_enqueue_style` to add your functionality to the Gutenberg editor.
+	 *
+	 * @since 0.4.0
+	 */
+	do_action( 'enqueue_block_editor_assets' );
```

Comment:
What if a plugin enqueue blocks but also register a plugin depending on `wp-edit-post` module?

My proposal here is to add a JavaScript actiion `registerBlocks` called on `domReach` (the WP package). This way no matter the script loading order, the blocks will always be registered at the right moment.

Reply:
What I had in mind is something like that:

```js
wp.domReady( () => {
    wp.hooks.doAction( 'registerBlocks' );
    wp.editPost.initializeEditor();
} );
```

And ask plugin authors to move their `registerBlockType` calls too the `registerBlocks` action. I know it might be a bit late for that, so the current method should work as well, and we could warn if `registerBlockType` is called before the hook.